include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -849,6 +849,7 @@ __pattern_scan_copy(_ExecutionPolicy&& __exec, _Iterator1 __first, _Iterator1 __
         __par_backend_hetero::__parallel_scan_copy(::std::forward<_ExecutionPolicy>(__exec), __buf1.all_view(),
                                                    __buf2.all_view(), __n, __create_mask_op, __copy_by_mask_op);
 
+    __res.event().wait();
```

Comment:
I think we don't need explicit wait, due to we call "get" below.
"get" does `get_host_access` in case of SYCL buffer and `sycl::event::wait_and_throw` in case USM.

`get_host_access` has implicit wait while a kernel is finalized:

SYCL spec tells:
```
As said in [Section 3.9.8](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:synchronization), the only blocking operations in SYCL (apart from explicit wait operations) are:

host accessor constructor, which waits for any kernels enqueued before its creation that write to the corresponding object to finish and be copied back to host memory before it starts processing. The host accessor does not necessarily copy back to the same host memory as initially given by the user;
```

Reply:
I believe you are right. The call to `get_host_access` seems like it should be sufficient synchronization in these cases.

However, there are test failures with the latest drivers/compiler where there appears to be a race when reading the USM output array. This means that perhaps the latest drivers have a bug that is causing the synchronization provided by `get_host_access` to not correctly work.

Reply:
"....when reading the USM output array" - in that case we do (must do, as we designed, at least) explicit wait before get a value from a future object. We have an non-sycl::buffer overload of `__wait_and_get_value`:
https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h#L582C5-L582C25

Or, probably, we are wrong with macro definition `ONEDPL_ALLOW_DEFERRED_WAITING`.
BTW, do we still need one?

Reply:
BTW 2,
'future' type has `wait()` methods as a part of contract (of public future API). we don't need call wait relying on 'event' API (which is returned by future::event() method)